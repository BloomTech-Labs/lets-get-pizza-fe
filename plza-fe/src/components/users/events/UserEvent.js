import React, { useState } from "react";
import { Icon, Item, Label } from "semantic-ui-react";
import Moment from "react-moment";
import EventUpdate from "./EventUpdate";

const UserEvent = ({ event, events, setEvents, deleteEvent }) => {
  const [toggleEdit, setToggleEdit] = useState(false);
  const [eventToEdit, setEventToEdit] = useState({});

  const editEvent = (event) => {
    setEventToEdit(event);
  };

  return (
    <>
      <Item>
        {/** Place Business Image Here **/}
        {/* <Item.Image src="http://place-puppy.com/250x200" /> */}

        <Item.Content>
          <Item.Header as="a" href={`/locations/${event.location_id}/events`}>
            {event.title}
          </Item.Header>
          <Item.Description>{event.description}</Item.Description>
          <Item.Meta>
            <span className="">{event.business_name}</span>
          </Item.Meta>
          <Item.Meta>
            <span className="">{event.address}</span>
          </Item.Meta>
          <Item.Extra>
            <Label title="Event date">
              <Icon name="calendar alternate" />
              <Moment format="ddd" date={event.start_time} />
              &#44;&#xa0;&#xa0;
              <Moment format="MMMM D" date={event.start_time} />
            </Label>

            <Label title="Event time">
              <Icon name="clock" />
              <Moment format="h:mm a" date={event.start_time} />
              &mdash;
              <Moment format="h:mm a" date={event.end_time} />
            </Label>

            <Label
              title="Edit event"
              as="a"
              onClick={() => {
                editEvent(event);
                setToggleEdit(!toggleEdit);
              }}
            >
              <Icon name="edit" />
            </Label>

            <Label
              title="Delete event"
              as="a"
              onClick={(e) => {
                e.stopPropagation();
                deleteEvent(event.id);
              }}
            >
              <Icon name="trash" />
            </Label>
          </Item.Extra>
        </Item.Content>
      </Item>
      {toggleEdit && (
        <EventUpdate
          event={event}
          events={events}
          setEvents={setEvents}
          eventToEdit={eventToEdit}
          setEventToEdit={setEventToEdit}
          toggleEdit={toggleEdit}
          setToggleEdit={setToggleEdit}
        />
      )}
    </>
  );
};

export default UserEvent;
