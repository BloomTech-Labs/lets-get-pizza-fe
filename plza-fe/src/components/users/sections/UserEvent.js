import React from "react";
import { Button, Icon, Image, Item, Label } from "semantic-ui-react";
import Moment from "react-moment";

const UserEvent = ({event}) => {
  console.log(event);
  return (
    <Item>
      <Item.Image src="http://place-puppy.com/250x200" />

      <Item.Content>
        <Item.Header as="a">{event.title}</Item.Header>
        <Item.Description>{event.description}</Item.Description>
        <Item.Meta>
          <span className="">Business Name</span>
        </Item.Meta>
        <Item.Meta>
          <span className="">Business Address</span>
        </Item.Meta>
        <Item.Extra>
          <Label title="Event date">
            <Icon name="calendar alternate" />
            <Moment format="ddd" date={event.start_time} />
            &#44;&#xa0;&#xa0;
            <Moment format="MMMM D" date={event.start_time} />
          </Label>

          <Label title="Event time">
            <Icon name="clock" />
            <Moment format="h:mm a" date={event.start_time} />
            &mdash;
            <Moment format="h:mm a" date={event.end_time} />
          </Label>
        </Item.Extra>
      </Item.Content>
    </Item>
  );
};

export default UserEvent;
