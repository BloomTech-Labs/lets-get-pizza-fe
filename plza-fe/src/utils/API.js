import axios from "axios";

/**
 * Creates an Axios instance with a predefined base URL as well as
 * sending the user's stored JWT as a header.
 */
export default axios.create({
  baseURL: process.env.REACT_APP_BASE_URL,
  headers: {
    Authorization: JSON.parse(localStorage.getItem("token")),
  },
  responseType: "json",
});
