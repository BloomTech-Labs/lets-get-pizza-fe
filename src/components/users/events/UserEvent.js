import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Icon, Item, Label } from "semantic-ui-react";
import Moment from "react-moment";
import EventUpdate from "./EventUpdate";
import { userDeleteEvent } from "../../../redux/actions/userActions";
import InviteModal from "../../items/InviteModal";

const UserEvent = ({ event }) => {
  const user = useSelector(state => state.user)
  const [toggleEdit, setToggleEdit] = useState(false);
  const dispatch = useDispatch();
  const [eventToEdit, setEventToEdit] = useState({});

  const editEvent = (event) => {
    setEventToEdit(event);
  };

  return (
    <>
      <Item>
        {/** Place Business Image Here **/}
        {/* <Item.Image src="http://place-puppy.com/250x200" /> */}

        <Item.Content>
          <Item.Header as="a" href={`/locations/${event.location_id}/events`}>
            {event.title}
          </Item.Header>
          <Item.Meta>Created By - You</Item.Meta>
          <Item.Description>{event.description}</Item.Description>
          <Item.Meta>
            <span className="">{event.business_name}</span>
          </Item.Meta>
          <Item.Meta>
            <span className="">{event.address}</span>
          </Item.Meta>
          <Item.Extra>
            <Label title="Event date">
              <Icon name="calendar alternate" />
              <Moment format="ddd" date={event.start_time} />
              &#44;&#xa0;&#xa0;
              <Moment format="MMMM D" date={event.start_time} />
            </Label>

            <Label title="Event time">
              <Icon name="clock" />
              <Moment format="h:mm a" date={event.start_time} />
              &mdash;
              <Moment format="h:mm a" date={event.end_time} />
            </Label>

            <Label
              title="Edit event"
              as="a"
              onClick={() => {
                editEvent(event);
                setToggleEdit(!toggleEdit);
              }}
            >
              <Icon name="edit" />
            </Label>

            <Label
              title="Delete event"
              as="a"
              onClick={(e) => {
                e.stopPropagation();
                dispatch(userDeleteEvent(event.id, user));
              }}
            >
              <Icon name="trash" />
            </Label>
            {user.id === event.user_id && <InviteModal event_id={event.id}/>}
          </Item.Extra>
        </Item.Content>
      </Item>
      {toggleEdit && (
        <EventUpdate
          event={event}
          events={user.events}
          eventToEdit={eventToEdit}
          setEventToEdit={setEventToEdit}
          toggleEdit={toggleEdit}
          setToggleEdit={setToggleEdit}
        />
      )}
    </>
  );
};

export default UserEvent;
