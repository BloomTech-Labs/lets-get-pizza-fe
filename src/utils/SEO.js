import React from 'react'
import { Helmet } from 'react-helmet'

const SEO = () => {
    return (
        <Helmet>
            {/* PAGE DESCRIPTION */}
            
            <meta charset="utf-8" />
            <title>Plza: Connection through pizza</title>
            <link rel="manifest" href="/manifest.json" />
            <link rel='icon' href='favicon.ico' type='image/png' size='16x16'/>
            <link rel="apple-touch-icon" href="logo192.png" />
            <meta name='description' content='Connecting pizza eaters & businesses- find & rate pizza and organize pizza-related social gatherings and events.' />

            {/* SOCIAL MEDIA META TAGS */}

            <meta property='og:title' content='Plza: Connection through pizza' />
            <meta property='og:type' content='website' />
            <meta property='og:description' content='Connecting pizza eaters & businesses- find & rate pizza and organize pizza-related social gatherings and events.' />
            <meta property='og:image' content='https://github.com/Lambda-School-Labs/lets-get-pizza-fe/blob/master/src/media/man-eating-pizza-3182799.jpg?raw=true'/>
            <meta property='og:url' content='https://pleazza.com' />
            <meta name='twitter:card' content='summary' />
            <meta name='twitter:title' content='Plza: Connection through pizza' />
            <meta name='twitter:description' content='Connecting pizza eaters & businesses- find & rate pizza and organize pizza-related social gatherings and events.' />
            
        </Helmet>
    )
}

export default SEO