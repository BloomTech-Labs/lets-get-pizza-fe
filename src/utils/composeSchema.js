import { object } from "yup";

/**
 * Compose Yup validation schema
 *
 * @param {object} initialSchema Example: `username`, `password`
 * @param {object} extraSchema
 * @returns {object} Composed object with initial and extra schema together
 */
export default function composeSchema(initialSchema, extraSchema) {
  return object().shape({ ...initialSchema, ...extraSchema });
}
