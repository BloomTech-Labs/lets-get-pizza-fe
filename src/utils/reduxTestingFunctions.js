// This function should take any component & an id
export const findElementById = (component, id) => {
  /*    FOR USE IN TESTING REDUX RENDERER   */
  // convert component to JSON so its easily managed with JavaScript
  const mother = component.toJSON();

  // Declare a variable to store current 'level' of looping
  // as we want to be able to loop as many times as the writer
  // has nested elements. The incoming component will always be a single
  // element, so its children must be nested in an array as if
  // we were accessing it from a parent.
  let currentLevel = [mother.children];

  // store the final item
  const filteredElement = [];

  // This function will dig into the arrays at the current level of nesting
  // and set the current level to that of its children.
  const repeatingLoop = () => {
    const updateLevel = [];
    currentLevel.forEach((arr) => {
      arr.forEach((object) => {
        if (object.children) {
          updateLevel.push(object.children);
        }
      });
    });
    currentLevel = updateLevel;
  };

  // This function will loop through all arrays of children at the current level
  // and will evaluate to see if any elements contain the given id from args.
  const lookForId = () => {
    currentLevel.forEach((arr) => {
      arr.forEach((object) => {
        if (object.props && object.props.id === id) {
          filteredElement.push(object);
        }
      });
    });
  };

  // This function will check to see if all children arrays are empty
  // in order to push an object with an error message so that the while
  // loop does not run forever if there is no id matching the id
  // recieved from args.
  const checkForStoppingPoint = () => {
    let combinedArr = 0;
    currentLevel.forEach((arr) => {
      if (arr.length !== 0) {
        combinedArr++;
      }
    });
    if (combinedArr === 0) {
      filteredElement.push({
        error: "no matching id found in the supplied component",
      });
    }
  };

  // Check to see if initial component object contains the desired id from args.
  if (mother.props.id === id) {
    return { ...mother.props, type: mother.type, children: mother.children };
  } else {
    // Otherwise run the functions to check if there are children,
    // if the current component contains the correct id, or if
    // the code needs to run at the next level of children.
    do {
      checkForStoppingPoint();
      lookForId();
      repeatingLoop();
      // do these functions until desired condition is met.
    } while (filteredElement.length === 0);
    return {
      ...filteredElement[0].props,
      type: filteredElement[0].type,
      children: filteredElement[0].children,
    };
  }
};

// When testing to see if a document contains text from props
// and I want to check that the data passed through correctly
// at least once, this function will find the first instance
export const findFirstProp = (textString, find, findAll) => {
  const regex = new RegExp(textString, "i");
  const array = findAll(regex);

  if (array.length <= 1) {
    return find(regex);
  } else {
    return array[0];
  }
};

/*    FOR USE IN TESTING REDUX ACTION CREATORS    */

// compares if items in an array are equal to each other
// requires an array of results & an array of expected
//      order of expectation array matters
// optional parameter of a single item not to be expected
export const compareExpectedCalls = (
  resultsArr,
  expectedArr,
  unexpectedItem
) => {
  resultsArr.forEach((result, idx) => {
    if (unexpectedItem) {
      expect(result).not.toEqual(unexpectedItem);
    }
    expect(result).toEqual(expectedArr[idx]);
  });
  expectedArr.forEach((expectation, idx) => {
    expect(expectation).toEqual(resultsArr[idx]);
  });
};

export const spreadCalls = (arrayOfArrays) => {
  let newArray = [];
  arrayOfArrays.forEach((array) => (newArray = [...newArray, ...array]));
    return newArray;
  };

export const compareExpectedState = (
  typeArray,
  initialState,
  payload,
  expectedState,
  reducer,
  failType
) => {
  // Map over given typeArray and return an array of action objects
  const actionsArray = typeArray.map((type) => ({ type, payload }));

  // Here we use .forEach() to run our test for each action in the array
  actionsArray.forEach((action) => {
    // Pass the initialState and action into the reducer
    // and set returnedState to the returned state from the reducer
    const returnedState = reducer(initialState, action);

    if (failType) {
      const failedState = reducer(initialState, { type: failType, payload });

      expect(failedState).not.toStrictEqual(expectedState);
    }

    expect(returnedState).toStrictEqual(expectedState);
    expect(returnedState).not.toStrictEqual(initialState);
    expect(returnedState).toBeDefined();
  });
};
