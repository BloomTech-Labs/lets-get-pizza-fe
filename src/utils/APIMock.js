import API from "./API";
import MockAdapter from "axios-mock-adapter";

export const APIMock = async (method, endpoint, resData) => {
  const url = endpoint
    ? `http://wwww.mockserver.com/${endpoint}`
    : "http://wwww.mockserver.com/";
  let mockAdapter = new MockAdapter(API);
  const response = resData
    ? {
        token: "test token",
        method: method,
        data: resData,
        endpoint: url
      }
    : {
        token: "test token",
        method: method,
      };

  if (method === "get") {
    mockAdapter.onGet(url).reply(200, response);
  } else if (method === "post") {
    mockAdapter.onPost(url).reply(201, response);
  } else if (method === "put") {
    mockAdapter.onPut(url).reply(200, response);
  } else if (method === "delete") {
    mockAdapter.onDelete(url).reply(200, response);
  }
  return mockAdapter.handlers[method][0].filter(
    (item) => typeof item === "object"
  )[0];
};
