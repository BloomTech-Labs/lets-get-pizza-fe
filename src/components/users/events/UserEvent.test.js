import React from "react";
import { useSelector } from "react-redux";
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";
import { findFirstProp } from "../../../utils/reduxTestingFunctions";
import UserEvent from "./UserEvent";

jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe("UserEvent", () => {
  const event = {
    title: "21st Birthday",
    description: "Come celebrate my 21st Birthday",
    start_time: "2020-01-20T08:00:00.000Z",
    end_time: "2020-01-22T08:00:00.000Z",
  };

  it("renders user with events", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        user: {
          username: "John",
          events: [
            {
              address:
                "500 Lakewood Center Mall, Lakewood (at Candlewood Ave), Lakewood, CA 90712, United States",
              business_name: "California Pizza Kitchen",
              description:
                "Come celebrate my 21st Birthday. Free pizzas and beers!",
              end_time: "2020-06-24T23:30:00.000Z",
              id: 4,
              location_id: 9,
              start_time: "2020-06-24T18:00:00.000Z",
              title: "Birthday Party!!!",
              user_id: 11,
            },
            {
              address: "69 Bompton Ave., Compton, CA 90221",
              business_name: "Kony Pizzerias",
              description: "test",
              end_time: "2020-06-26T23:40:00.000Z",
              id: 106,
              location_id: 30,
              start_time: "2020-06-26T18:00:00.000Z",
              title: "test user add event redux hooks",
              user_id: 11,
            },
          ],
        },
      })
    );
    const { container } = render(<UserEvent event={event}/>);
    expect(container).toMatchSnapshot();
  });


  it("renders user without events", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        user: {
          username: "John",
          events: [],
        },
      })
    );
    const { container } = render(<UserEvent event={event}/>);
    expect(container).toMatchSnapshot();
  });

  it("renders text and values from props", () => {
    const { getByText, getAllByText } = render(<UserEvent event={event}/>);

    const title = findFirstProp("21st Birthday", getByText, getAllByText);
    
  });
});
