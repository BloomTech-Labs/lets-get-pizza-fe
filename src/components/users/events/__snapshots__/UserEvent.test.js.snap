// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEvent renders user with events 1`] = `
<div>
  <div
    class="item"
  >
    <div
      class="content"
    >
      <a
        class="header"
        href="/locations/undefined/events"
      >
        21st Birthday
      </a>
      <div
        class="meta"
      >
        Created By - You
      </div>
      <div
        class="description"
      >
        Come celebrate my 21st Birthday
      </div>
      <div
        class="meta"
      >
        <span
          class=""
        />
      </div>
      <div
        class="meta"
      >
        <span
          class=""
        />
      </div>
      <div
        class="extra"
      >
        <div
          class="ui label"
          title="Event date"
        >
          <i
            aria-hidden="true"
            class="calendar alternate icon"
          />
          <time
            datetime="1579507200000"
          >
            Mon
          </time>
          ,  
          <time
            datetime="1579507200000"
          >
            January 20
          </time>
        </div>
        <div
          class="ui label"
          title="Event time"
        >
          <i
            aria-hidden="true"
            class="clock icon"
          />
          <time
            datetime="1579507200000"
          >
            12:00 am
          </time>
          —
          <time
            datetime="1579680000000"
          >
            12:00 am
          </time>
        </div>
        <a
          class="ui label"
          title="Edit event"
        >
          <i
            aria-hidden="true"
            class="edit icon"
          />
        </a>
        <a
          class="ui label"
          title="Delete event"
        >
          <i
            aria-hidden="true"
            class="trash icon"
          />
        </a>
        <a
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="add user icon"
          />
          Invite Friends
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserEvent renders user without events 1`] = `
<div>
  <div
    class="item"
  >
    <div
      class="content"
    >
      <a
        class="header"
        href="/locations/undefined/events"
      >
        21st Birthday
      </a>
      <div
        class="meta"
      >
        Created By - You
      </div>
      <div
        class="description"
      >
        Come celebrate my 21st Birthday
      </div>
      <div
        class="meta"
      >
        <span
          class=""
        />
      </div>
      <div
        class="meta"
      >
        <span
          class=""
        />
      </div>
      <div
        class="extra"
      >
        <div
          class="ui label"
          title="Event date"
        >
          <i
            aria-hidden="true"
            class="calendar alternate icon"
          />
          <time
            datetime="1579507200000"
          >
            Mon
          </time>
          ,  
          <time
            datetime="1579507200000"
          >
            January 20
          </time>
        </div>
        <div
          class="ui label"
          title="Event time"
        >
          <i
            aria-hidden="true"
            class="clock icon"
          />
          <time
            datetime="1579507200000"
          >
            12:00 am
          </time>
          —
          <time
            datetime="1579680000000"
          >
            12:00 am
          </time>
        </div>
        <a
          class="ui label"
          title="Edit event"
        >
          <i
            aria-hidden="true"
            class="edit icon"
          />
        </a>
        <a
          class="ui label"
          title="Delete event"
        >
          <i
            aria-hidden="true"
            class="trash icon"
          />
        </a>
        <a
          class="ui label"
        >
          <i
            aria-hidden="true"
            class="add user icon"
          />
          Invite Friends
        </a>
      </div>
    </div>
  </div>
</div>
`;
